src/clients/mock_tiled_client.rs

```diff
@@ -1,17 +1,28 @@
 use std::fs::File;
 use std::path::Path;
 
+use serde::de::DeserializeOwned;
+
 use crate::clients::{Client, ClientResult};
 use crate::schemas::tiled_metadata::Metadata;
-pub struct MockTiledClient;
+pub struct MockTiledClient {
+    pub dir_path: String,
+}
 
-impl Client for MockTiledClient {
-    async fn metadata(&self) -> ClientResult<Metadata> {
+impl MockTiledClient {
+    async fn load_file_into_struct<T: DeserializeOwned>(&self, file: &str) -> ClientResult<T> {
         println!("Requesting data from mock");
 
-        let path = Path::new("./resources/tiled_metadata.json");
+        let path_str = self.dir_path.to_string() + file;
+        let path = Path::new(&path_str);
         let file = File::open(&path)?;
 
         Ok(serde_json::from_reader(file)?)
+
+    }
+}
+impl Client for MockTiledClient {
+    async fn metadata(&self) -> ClientResult<Metadata> {
+        self.load_file_into_struct::<Metadata>(&"tiled_metadata.json").await
```

Comment:
String literals are already `&str`. And I think the types can be inferred here

```suggestion
        self.load_file_into_struct("tiled_metadata.json").await
```